pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -30,6 +30,192 @@ import (
+			"password"))
+
+		oldEmailBus := bus.GetHandlerCtx("SendEmailCommandSync")
+		mockEmails := &mockEmailHandler{}
+		bus.AddHandlerCtx("", mockEmails.sendEmailCommandHandlerSync)
+		t.Cleanup(func() {
+			bus.AddHandlerCtx("", oldEmailBus)
+		})
+
+		testReceiversURL := fmt.Sprintf("http://grafana:password@%s/api/alertmanager/grafana/config/api/v1/receivers/test", grafanaListedAddr)
+		// nolint:gosec
+		resp, err := http.Post(testReceiversURL, "application/json", strings.NewReader(`{
+	"receivers": [{
+		"name":"receiver-1",
+		"grafana_managed_receiver_configs": [
+			{
+				"uid":"",
+				"name":"receiver-1",
+				"type":"email",
+				"disableResolveMessage":false,
+				"settings":{
+					"addresses":"example@email.com"
+				},
+				"secureFields":{}
+			}
+		]
+	}]
+}`))
+		require.NoError(t, err)
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+		t.Cleanup(func() {
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+
+		var result apimodels.TestReceiversResult
+		require.NoError(t, json.NewDecoder(resp.Body).Decode(&result))
+
+		require.Len(t, result.Receivers, 1)
+		require.Len(t, result.Receivers[0].Configs, 1)
+		require.Equal(t, apimodels.TestReceiversResult{
+			Receivers: []apimodels.TestReceiverResult{{
+				Name: "receiver-1",
+				Configs: []apimodels.TestReceiverConfigResult{{
+					Name:   "receiver-1",
+					UID:    result.Receivers[0].Configs[0].UID,
+					Status: "ok",
+				}},
+			}},
+			NotifedAt: result.NotifedAt,
+		}, result)
+
+		require.Len(t, mockEmails.emails, 1)
+		require.Equal(t, []string{"example@email.com"}, mockEmails.emails[0].To)
+	})
+
+	t.Run("assert invalid receiver returns 400 Bad Request", func(t *testing.T) {
+		// Setup Grafana and its Database
+		dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+			EnableFeatureToggles: []string{"ngalert"},
+		})
+		store := testinfra.SetUpDatabase(t, dir)
+		store.Bus = bus.GetBus()
+		grafanaListedAddr := testinfra.StartGrafana(t, dir, path, store)
+		require.NoError(t, createUser(t,
+			store,
+			models.ROLE_EDITOR,
+			"grafana",
+			"password"))
+
+		oldEmailBus := bus.GetHandlerCtx("SendEmailCommandSync")
+		mockEmails := &mockEmailHandler{}
+		bus.AddHandlerCtx("", mockEmails.sendEmailCommandHandlerSync)
+		t.Cleanup(func() {
+			bus.AddHandlerCtx("", oldEmailBus)
+		})
+
+		testReceiversURL := fmt.Sprintf("http://grafana:password@%s/api/alertmanager/grafana/config/api/v1/receivers/test", grafanaListedAddr)
+		// nolint:gosec
+		resp, err := http.Post(testReceiversURL, "application/json", strings.NewReader(`{
+	"receivers": [{
+		"name":"receiver-1",
+		"grafana_managed_receiver_configs": [
+			{
+				"uid":"receiver-1-config-1",
+				"name":"receiver-1",
+				"type":"email",
+				"disableResolveMessage":false,
+				"settings":{},
+				"secureFields":{}
+			}
+		]
+	}]
+}`))
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, resp.Body.Close())
+		})
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
```

Comment:
```suggestion
		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
		require.JSONEq(t, string(b), "{\"message\":\"unknown receiver: receiver-1-config-1\"}")

```

Can we also assert on the error body to make sure it reads correct?